lib/adapters/file-adapter.js

```diff
@@ -56,10 +72,28 @@ var Read = Juttle.proc.source.extend({
     start: function() {
         var self = this;
         var stream = this.fetch();
+        var last;
 
         self.parser.parseStream(stream, function(points) {
+            var validPoints = [];
             self.parseTime(points, self.timeField);
-            self.emit(points);
+            _.each(points, function(point) {
+                if (last && last.gt(point.time)) {
+                    // out of order points are dropped
+                    self.trigger('warning', self.runtime_error('RT-OUT-OF-ORDER'), {
+                        proc: 'read file'
+                    });
+                } else {
+                    // timeless points just pass through at this point
+                    if (point.time && point.time.moment) {
```

Comment:
Yea, that `values` is our [library](https://github.com/juttle/juttle/blob/c3f5ab745e2123ac2f040f02d16b0e1f81006e37/lib/runtime/values.js#L48-L56).